### ISSUE TITLE
FR: open wikipedia in external (browser)

### ISSUE THREAD

**Issue by farfromrefug** [koreader/koreader#12974]
**Does your feature request involve difficulty completing a task? Please describe.**
I am developing/maintaining a fork of Aard2 app which supports offline wikipedia.

Right now i cant use the slob data i have in aard2 with koreader because koreader queries/opens wikipedia internally

**Describe the solution you'd like**
Add a setting to open/search wikipedia using external ( kind of like the dictionary). That way it would open in any browser app.
It would also work with Aaard2 which handles wikipedia URLs

**Comment by Frenzie** [comment]
Assuming you're talking about Android, that's already covered by  https://github.com/koreader/koreader/wiki/Android-tips-and-tricks#customize-dictionary-list

**Comment by Frenzie** [comment]
In light of #12973 this doesn't make too much sense to me (just switch to a browser?) but pinging @pazos for a second opinion.

**Comment by farfromrefug** [comment]
@Frenzie thanks that indeed solve the issue about using my custom fork of aard2. However would still be best to use the lookup intent sonthst everyone can enjoy it. This would require each user to add it ( and i really intend to manake my fork and release IR on fdroid ).

But I dont think it solves the Wikipedia issue (this one). I would like to be able to use external Wikipedia lookup . could be the official Wikipedia app or aard. On your side it simply means adding a way to open the URL "http://en.wikipedia.org...." for wikipedia lookup. That way android can handle it. 
Not sure If that is clear end enough

**Comment by pazos** [comment]
On android opening a link in a browser should prompt you about the browser you want to use each time (unless you setup one by default).

The same prompting for each intent when using a 3rdparty dictionary.

If I understood you you want external links which belong to wikipedia domain to be resolved by opening any app that could resolv its url. Is that right?

It looks like easy solvable with an userpatch with some pattern matching and using `android.openLink()` with the url.

**Comment by farfromrefug** [comment]
@pazos yes i would expect to be able to use external app for wikipedia just like you do with dictionary.
Ok i get it could be solvable with a userpatch. Would still be awesome to have it included for everyone as it would allow offline wikipedia access

**Comment by pazos** [comment]
As far as I am concerned I'm fine with that change :)

frontend/device/thirdparty.lua is compatible with dictionary and translator (albeit the last one is not implemented yet). It needs a new category "wikipedia"

You can repurpose the same logic used by EXTERNAL_DICTS and implement a hook somewhere before wikipedia UIs.

Also the same exact thing but for translator apps is also very welcome.

The only thing missing is a voluntary. Lets keep this ticket open while we wait ...

**Comment by farfromrefug** [comment]
@pazos ok might look at it in the future if i can. Though might not be before September 2025 :P In the meantime as i might quickly need this before i am out of computer, i might got the userpatch way. Thank you!

**Comment by pazos** [comment]
If you do the userpatch thing please share here. It will be useful for many others :) We try to document them https://github.com/koreader/koreader/issues?q=+is%3Aissue+label%3A%22User+patch+available%22+
